core/src/price_finding/optimization_price_finder.rs

```diff
@@ -193,28 +193,59 @@ pub struct OptimisationPriceFinder {
     io_methods: Arc<dyn Io + Send + Sync>,
     fee: Option<Fee>,
     solver_type: SolverType,
-    price_oracle: Box<dyn PriceEstimating + Sync>,
-    min_avg_fee_per_order: u128,
+    price_oracle: Box<dyn PriceEstimating + Send + Sync>,
+    gas_station: Arc<dyn GasPriceEstimating + Send + Sync>,
+    min_avg_fee_subsidy_factor: f64,
+    default_min_avg_fee_per_order: u128,
     internal_optimizer: InternalOptimizer,
 }
 
 impl OptimisationPriceFinder {
     pub fn new(
         fee: Option<Fee>,
         solver_type: SolverType,
-        price_oracle: impl PriceEstimating + Sync + 'static,
-        min_avg_fee_per_order: u128,
+        price_oracle: impl PriceEstimating + Send + Sync + 'static,
+        gas_station: Arc<dyn GasPriceEstimating + Send + Sync>,
+        min_avg_fee_subsidy_factor: f64,
+        default_min_avg_fee_per_order: u128,
         internal_optimizer: InternalOptimizer,
     ) -> Self {
         OptimisationPriceFinder {
             io_methods: Arc::new(DefaultIo),
             fee,
             solver_type,
             price_oracle: Box::new(price_oracle),
-            min_avg_fee_per_order,
+            gas_station,
+            min_avg_fee_subsidy_factor,
+            default_min_avg_fee_per_order,
             internal_optimizer,
         }
     }
+
+    async fn compute_min_average_fee_per_order(&self) -> u128 {
+        let eth_owl_xrate = match self.price_oracle.get_eth_price().await {
+            Some(owl_price) => (owl_price as f64) / 10f64.powi(18),
+            None => {
+                log::warn!("failed to find ETH price estimate");
+                return self.default_min_avg_fee_per_order;
```

Comment:
How about not requiring a min avg fee in this very rare case?

Reply:
This was more to keep the current behaviour until #1200 is implemented. At that point we can take it out. I guess when we can't get gas price estimates then it will be hard to reliably send a transaction anyway.

Reply:
I think this is now nicely implemented using the `MinAverageFeeComputing` abstraction where we have the default implementation that computes it based on the gas estimate and price estimate as well as a `FixedMinAverageFee` implementation and a `PriorityMinAverageFee` to combine them. This way we can fall back to using a fixed value if we want to, or remove the fixed value all together by changing how we initialize in `driver/src/main.rs`.